Sources/Purchasing/Purchases/PurchasesOrchestrator.swift

```diff
@@ -1107,6 +1114,32 @@ private extension PurchasesOrchestrator {
             self.presentedOfferingIDsByProductID.modify { $0[productIdentifier] = package.offeringIdentifier }
         }
     }
+
+    /// Computes a `ProductRequestData` for an active subscription found in the receipt,
+    /// or `nil` if there is any issue fetching it.
+    func createProductRequestData(
+        with receiptData: Data,
+        completion: @escaping (ProductRequestData?) -> Void
+    ) {
+        guard let receipt = try? self.receiptParser.parse(from: receiptData) else { completion(nil); return }
+
+        // Find the latest purchased subscription
+        guard let productIdentifier = receipt.inAppPurchases
+            .lazy
+            .filter({ $0.isActiveSubscription })
+            .sorted(by: { $0.purchaseDate > $1.purchaseDate })
+            .first?
+            .productId else { completion(nil); return }
```

Comment:
this feels out of scope for this class, tbh. Like there should either be a separate object for it, or potentially the ReceiptParser itself should be able to answer the latest subscriptions found in the receipt. 

Also... This shouldn't be limited to subscriptions, we're really just looking for a set of all product ids the customer has ever purchased

Reply:
> this feels out of scope for this class, tbh. Like there should either be a separate object for it, or potentially the ReceiptParser itself should be able to answer the latest subscriptions found in the receipt.

Totally good point

> This shouldn't be limited to subscriptions

I can remove that filter, but....

> we're really just looking for a set of all product ids the customer has ever purchased

But we need to pick one, because we post only 1 `ProductRequestData` based on a single `StoreProduct`.

Reply:
ohh interesting. We should check with the backend team, maybe this is a limitation in khepri that we can work around. If not, then I think the active subscription approach is as good as any

Reply:
We do not support multiple productIds by receipt. I think there are a lot of things that would have to change to support more than one product. [Receipt](https://github.com/RevenueCat/khepri/blob/main/khepri/models/receipt.py#L106) and [Transaction](https://github.com/RevenueCat/khepri/blob/main/khepri/models/apple_transaction.py#L67) are laid out to have a single product.
There was talk also to support multi lineItem receipts, but I don't think we have a concrete plan for this.

Reply:
@swehner that's good to know. We're not really looking to map this to a purchase of multiple products, really, we're just trying to provide price information for the products that this user has purchased in the past that they might be restoring. Would that be achievable _without_ trying to convert this from a restore event to a purchase event/

Reply:
@swehner @antoniobg could I get final confirmation that sending the price of the active subscription + the product id in a restore won't negatively affect any endpoints / webhooks?

Reply:
@aboedo I think it should be fine. It might affect the prices for newer transactions for the receipts for which we didn't have cached prices before, but I think it will be more accurate